src/main/java/io/github/fripe070/ghosthunting/item/LightSwitchItem.java

```diff
@@ -14,16 +22,38 @@ public LightSwitchItem(Block block, Settings settings) {
 
     BlockPos target_block = null;
 
+    @Override
     public ActionResult useOnBlock(ItemUsageContext context) {
-        ActionResult actionResult;
-        if (target_block == null) {
-            target_block = context.getBlockPos();
-            context.getWorld().setBlockState(target_block, Blocks.GRAY_CONCRETE.getDefaultState());
-
-            actionResult = ActionResult.SUCCESS;
-        } else {
-            actionResult = super.useOnBlock(context);
-            target_block = null;
+        ActionResult actionResult = ActionResult.PASS;
+        if (!context.getWorld().isClient()) {
+            if (target_block == null) {
+                target_block = context.getBlockPos();
+                context.getWorld().setBlockState(target_block, Blocks.GRAY_CONCRETE.getDefaultState());
+
+                actionResult = ActionResult.SUCCESS;
+            } else {
+                actionResult = super.useOnBlock(context);
+
+                BlockPos clickPos = context.getBlockPos();
+
+                switch (context.getSide()) {
+                    case NORTH -> clickPos = clickPos.north(1);
+                    case SOUTH -> clickPos = clickPos.south(1);
+                    case EAST -> clickPos = clickPos.east(1);
+                    case WEST -> clickPos = clickPos.west(1);
+                }
```

Comment:
What does this actually do? And why is it lacking any caases for up/down?

Reply:
Up/down will be added tonight. Fixed NBT being placed on the block the switch is placed on